kubert/src/lease.rs

```diff
@@ -463,11 +469,52 @@ impl LeaseManager {
         })
     }
 
-    fn is_conflict(err: &Error) -> bool {
-        matches!(
-            err,
-            Error::Api(kube_client::Error::Api(kube_core::ErrorResponse { code, .. }))
-                if hyper::StatusCode::from_u16(*code).ok() == Some(hyper::StatusCode::CONFLICT)
-        )
+    fn should_resync(err: &Error) -> bool {
+        match err {
+            Error::Api(kube_client::Error::Api(kube_core::ErrorResponse { code, .. })) => {
+                let status = hyper::StatusCode::from_u16(*code).ok();
+
+                // A 409 Conflict indicates that another process updated the
+                // claim's resource version, while a 410 Gone indicates that our
+                // resource version is too old. In re-sync the state and try
+                // again.
+                status == Some(hyper::StatusCode::CONFLICT)
+                    || status == Some(hyper::StatusCode::GONE)
```

Comment:
looking at the [`kube_client` docs](https://docs.rs/kube-client/latest/kube_client/enum.Error.html#variant.Api) I noticed that a `410 Gone` may also be returned in the case that a resource version is out of date, and figured that we should also re-sync in that case. Happy to back out that part of the change if this isn't correct.